athena-vertica/src/main/java/com/amazonaws/connectors/athena/vertica/VerticaJdbcSplitQueryBuilder.java

```diff
@@ -0,0 +1,297 @@
+                ", Compression='snappy'\n" +
+                ", fileSizeMB=1000) \n" +
+                "AS \n";
+
+        sql.append(exportSqL);
+
+        //get the column names to be queried
+        String columnNames = tableSchema.getFields().stream()
+                .map(Field::getName)
+                .map(this::quote)
+                .collect(Collectors.joining(", "));
+
+        sql.append("SELECT ");
+        sql.append(columnNames);
+
+        if (columnNames.isEmpty()) {
+            sql.append("null");
+        }
+
+        sql.append(getFromClauseWithSplit(schema, table));
+
+        HashMap<String,TypeAndValue> accumulator = new HashMap<>();
+
+        List<String> clauses = toConjuncts(tableSchema.getFields(), constraints, accumulator);
+
+        if (!clauses.isEmpty()) {
+            sql.append(" WHERE ")
+                    .append(Joiner.on(" AND ").join(clauses));
+        }
+
+        //wrap the export around the generated sql
+        LOGGER.info("Generated SQL without the constraints : {}", sql.toString());
+
+        //Using StringTemplates to fill in the values of constraints
+        ST sqlTemplate = new ST(sql.toString());
+
+        for (Map.Entry<String,TypeAndValue> entry : accumulator.entrySet()){
+
+            TypeAndValue typeAndValue = entry.getValue();
+            Types.MinorType minorTypeForArrowType = Types.getMinorTypeForArrowType(typeAndValue.getType());
+            String colName = entry.getKey();
+            //TODO: Need to review this and add other data types here!!
+            switch (minorTypeForArrowType) {
+                case INT:
+                    sqlTemplate.add(colName, typeAndValue.getValue());
+                    break;
+
+                case VARCHAR:
+                    String val = "'" + typeAndValue.getValue() + "'";
+                    sqlTemplate.add(colName, val);
+                    break;
+
+                case DECIMAL:
+                    ArrowType.Decimal decimalType = (ArrowType.Decimal) typeAndValue.getType();
+                    sqlTemplate.add(colName, typeAndValue.getValue());
+                    break;
+
+                default:
+                    throw new UnsupportedOperationException(String.format("Can't handle type: %s, %s", typeAndValue.getType(), minorTypeForArrowType));
+            }
+
+
+        }
+
+        String sqlStatement = sqlTemplate.render();
+        LOGGER.info(sqlStatement);
+
+        return sqlStatement;
+    }
+
+   // protected abstract String getFromClauseWithSplit(final String schema, final String table);
+
+    protected String getFromClauseWithSplit(String schema, String table)
+    {
+        StringBuilder tableName = new StringBuilder();
+        if (!Strings.isNullOrEmpty(schema)) {
+            tableName.append(quote(schema)).append('.');
+        }
+        tableName.append(quote(table));
+
+        return String.format(" FROM %s ", tableName);
```

Comment:
This is something we should be using StringTemplate for.

Reply:
used StringTemplates